content/posts/irods-development-update-december-2024.md

```diff
@@ -0,0 +1,26 @@
+Title: iRODS Development Update: December 2024
+Date: 2024-12-19 12:00
+Author: Kory Draughn
+Slug: irods-development-update-december-2024
+Status: published
+
+
+Hello reader,
+
+I can't believe we're at the end of December, which means the end of 2024! Wow, time really does fly. Anyway, you know what time it is, so let's jump right into it.
+
+Starting with the server, work to redesign the startup/shutdown logic for iRODS 5 is complete and has been merged. This is exciting because it brings the server closer to what administrators expect from a modern server application. With the startup/shutdown code merged, work to overhaul support for systemd has begun. A new GenQuery1 parser has been merged into the iRODS 5 implementation as well. The parser resolves all long-standing string parsing issues involving whitespace, single quotes, etc. We originally planned to introduce the parser into iRODS 4, but that plan was discarded to avoid potentially breaking existing deployments. There's been some issues reported about multi-byte characters recently, so we've started investigating improvements around unicode. `iadmin` now returns non-zero error codes appropriately. A number of things have recently been marked as deprecated in the iRODS 4 series, e.g. `imeta qu`, the interactive mode of `imeta`, and the locking/unlocking APIs for data objects. `ilocate` and `igetwild` are also on the road to deprecation and removal as well.
```

Comment:
```suggestion
Starting with the server, work to redesign the startup/shutdown logic for iRODS 5 is complete and has been merged. This is exciting because it brings the server closer to what administrators expect from a modern server application. With the startup/shutdown code merged, work to overhaul support for systemd has begun. A new GenQuery1 parser has been merged into the iRODS 5 implementation as well. The parser resolves all long-standing string parsing issues involving whitespace, single quotes, etc. We originally planned to introduce the parser into iRODS 4, but that plan was discarded to avoid potentially breaking existing deployments. There have been some issues reported about multi-byte characters recently, so we've started investigating improvements around unicode. `iadmin` now returns non-zero error codes appropriately. A number of things have recently been marked as deprecated in the iRODS 4 series, e.g. `imeta qu`, the interactive mode of `imeta`, and the original locking/unlocking APIs for data objects. `ilocate` and `igetwild` are also on the road to deprecation and removal as well.
```